### PR TITLE
Add stack termination protection to bigmac streams

### PR DESCRIPTION
**Pull request by benmartin-coforma**
### Description
Stack termination protection is a [serverless plugin](https://www.serverless.com/plugins/serverless-stack-termination-protection) to use an [AWS feature](https://docs.aws.amazon.com/emr/latest/ManagementGuide/UsingEMR_TerminationProtection.html) that prevents long-running clusters from being accidentally terminated. They still can be, but you have to remove the protection first.

Across all apps managed by our team (SEDS, CARTS, QMR, MCR, MFP), there are 37 `serverless.yml` files for defining AWS stacks. Of those, 36 already include `serverless-stack-termination-protection`. `carts-bigmac-streams` is the 37th. If termination protection is a good idea for all our other stacks, I think we want it for this one as well.

### Related ticket(s)
n/a

---
### How to test
1. Log in to AWS
2. Follow the steps described [here](https://docs.aws.amazon.com/emr/latest/ManagementGuide/UsingEMR_TerminationProtection.html#emr-termination-protection-running-cluster) to confirm that protection is enabled

### Important updates
n/a

---
### Author checklist

- [x] I have performed a self-review of my code
- ~[ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary~
- ~[ ] I have updated relevant documentation, if necessary~

### CODE REVIEW

**Review by BearHanded — DISMISSED** [review]

File: services/carts-bigmac-streams/serverless.yml [hunk 1]
```diff
@@ -12,10 +12,21 @@ plugins:
...
   region: ${opt:region, self:provider.region}
+  serverlessTerminationProtection:
+    stages: # This is a list of common names for important envs that should not be destroyed. You can remove the stage names your project doesn't use; this list is meant to be inclusive.
+      - master
+      - val
+      - production
+      - develop
+      - main
+      - impl
+      - val
```
Comment (trancjessica) [inline services/carts-bigmac-streams/serverless.yml:28]: [nit, non-blocking] duplicate from line 23? :-)

**Review by trancjessica — DISMISSED** [review]
i cannot access our AWS accounts (yet) so I didn't pull this in and deploy. However, lookeda t the serverless docs and this looks good to me, minus the one nit comment

File: services/carts-bigmac-streams/serverless.yml (hunk 1 above)
Comment (benmartin-coforma) [reply services/carts-bigmac-streams/serverless.yml]: Right you are! Removed.

**Review by gmrabian — APPROVED** [review]
CARTS only has `main`, `master`, `val`, and `production` so we could simplify this list, but no big deal

**Review by BearHanded — APPROVED** [review]

**Pull request merged**